gem/lib/pupperware/spec_helper.rb

```diff
@@ -61,17 +63,19 @@ def run_command(env = ENV.to_h, command, stream: STDOUT)
     { status: status, stdout: stdout_string, stderr: stderr_string }
   end
 
-  def retry_block_up_to_timeout(timeout, exit_early_on_error_type: [], raise_custom_error_type: nil, &block)
+  def retry_block_up_to_timeout(timeout, exit_early_on: -> e { false }, raise_custom_error_type: nil, &block)
```

Comment:
Can you throw a docstring on this method that shows an example argument for `exit_early_on` ? 

Or maybe just say that this is an "anonymous function" or whatever so I can google for Ruby examples of that?

Never seen that syntax before!

Reply:
They're informally referred to as stabby lambdas, part of Ruby 1.9.

https://isotope11.com/blog/stabby-lambda

Reply:
I mean for the next user of this code, not for this PR review :)

Maybe say what the function parameters and return value (if any) should be, and that it's a "stabby lambda" so they can google for that?